index.html

```diff
@@ -1319,6 +1323,40 @@ <h3><dfn>close()</dfn> method</h3>
       <li>Return |promise|.
     </ol>
   </section>
+  <section>
+    <h3><dfn>forget()</dfn> method</h3>
+
+    <aside class="example">
+      It is posssible to voluntarily revoke a permission to a serial port that was granted by a user.
+
+      <pre class="js">
+        // Request a serial port.
+        const port = await navigator.serial.requestPort();
+
+        // Then later... revoke permission to the serial port.
+        await port.forget();
+      </pre>
+    </aside>
+
+    The {{SerialPort/forget()}} method steps are:
+
+    <ol>
+      <li>Let |promise| be [=a new promise=].
+      <li>Set [=this=].{{SerialPort/[[state]]}} to `"forgetting"`.
+      <li>
+        If [=this=].{{SerialPort/[[readable]]}} is not `null`, invoke
+        [=ReadableStream/cancel=] on [=this=].{{SerialPort/[[readable]]}}.
+      <li>
+        If [=this=].{{SerialPort/[[writable]]}} is not `null`, invoke
+        [=WritableStream/abort=] on [=this=].{{SerialPort/[[writable]]}}.
+      <li>
+        Remove [=this=] from the sequence of available serial ports on
+        the system which the user has allowed the site to access as the
+        result of a previous call to {{Serial/requestPort()}}.
+      <li>Set [=this=].{{SerialPort/[[state]]}} to `"forgotten"`.
```

Comment:
The specification should probably mention the possibility that forgetting a device is not possible (e.g. permission was granted by administrator policy). Whether in that case the device is still put into the `"forgotten"` state is an interesting question. I think it probably shouldn't be because it'll be weird for `getPorts()` to be returning a `SerialPort` instance that is otherwise unusable. Alternatively we could keep track of forgotten ports and hide them until the page reloads. That seems more complex though perhaps less surprising for the developer. What do you think?

Reply:
Good catch indeed! That's the first time I see administrator policy mentioned in a spec and it makes sense there!

Reply:
I think we want to make failure silent in this case. A site can discover that the device can't be forgotten by observing the side effects of this method but I don't think that's something that should be considered an error.

Reply:
Done